From b7a658efa5c9d9ef94a484ec88f94e5a1f68fc52 Mon Sep 17 00:00:00 2001
From: Niall Pemberton <niallp@apache.org>
Date: Thu, 1 Oct 2009 11:19:24 +0000
Subject: [PATCH] BEANUTILS-327 JDBCDynaClass throws class not found exception
 for arrays (see http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=6516909) -
 thanks to Sascha Riemann for the patch

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/beanutils/trunk@820611 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/beanutils/JDBCDynaClass.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/java/org/apache/commons/beanutils/JDBCDynaClass.java b/src/java/org/apache/commons/beanutils/JDBCDynaClass.java
index 9769e17b8..0ca825a4b 100644
--- a/src/java/org/apache/commons/beanutils/JDBCDynaClass.java
+++ b/src/java/org/apache/commons/beanutils/JDBCDynaClass.java
@@ -159,7 +159,8 @@ protected Class loadClass(String className) throws SQLException {
             if (cl == null) {
                     cl = this.getClass().getClassLoader();
             }
-            return (cl.loadClass(className));
+            // use Class.forName() - see BEANUTILS-327
+            return Class.forName(className, false, cl);
         } catch (Exception e) {
             throw new SQLException(
                     "Cannot load column class '" + className + "': " + e);
